std/array.d

```diff
@@ -3519,6 +3519,41 @@ if (isDynamicArray!A)
         return cast(typeof(return))(_data ? _data.arr : null);
     }
 
+    /**
+     * Returns: The current length of the array
+     */
+    @property size_t length() const
+    {
+        return _data ? _data.arr.length : 0U;
+    }
+
+    /**
+     * Set a new length for the array
+     * Params:
+     *    a = The new length
+     */
+    @property void length(size_t newLength)
+    {
+        if (_data)
+        {
+            if (newLength > _data.arr.length)
+            {
+                size_t oldLength = _data.arr.length;
+                reserve(newLength - oldLength);
+                _data.arr = _data.arr.ptr[0 .. newLength];
+                _data.arr[oldLength - 1 .. newLength] = T.init;
```

Comment:
Shouldn't this be `emplace` and not assignment?

Reply:
I don't think it is possible because you need to repeat `T.init` times `newLength-oldLength`, but can you suggest a change? This needs to be done similarly to `_d_arraysetlengthT` and `_d_arraysetlengthiT` runtime hooks. I'm doing a change locally to do that.